### PR TITLE
feat(letters): Various enhancements

### LINKED ISSUE #160: [FEAT] Letters cog enhancements
# Overview
## Describe the solution you'd like

These changes are all tied to the letters cog
- if resulting message is too long (> 2000 chars) it should pagify the output or not output anything at all
- option to output without the code block/only the code block
- maybe expand the accepted characters to include punctuation? Have to be careful to only accept certain punctuation here in case of ping escaping

## Is this suggestion based on an existing problem? If so, please describe it

The bot errors out if resulting message is > 2000 chars instead of failing gracefully

# DoD
- Fail gracefully if resulting message is > 2000 chars OR pagify output

# Stretch
- `--raw` and `--without-raw` options in command (flag naming subject to change)
- Accept some punctuation in command (maybe `!?,.'`?)

### PR DESCRIPTION
**Pull request by tigattack**
# Initial Checklist
- [x] Has @tigattack / @kdavis / @issy been added as a reviewer?
- [x] If applicable, have the relevant project(s), milestone(s), and label(s) been applied?
- [x] If applicable, have you added details of the cog to the readme as per [README.md](https://github.com/rHomelab/LabBot-Cogs/blob/develop/README.md#cog-summaries)?

# Details
* **Does this resolve an issue?**
Resolves #160

### New Features
* Add limited support for special characters (!?#'.,).
* Add `-raw` flag to output emote code in code block.
* Limit output length to 2000 characters.

### Breaking Changes
N/A.

## Additional
N/A.

### CODE REVIEW

File: letters/letters.py [hunk 1]
```diff
@@ -12,41 +15,64 @@ class Letters(commands.Cog):
...
                 letters += f"{nums[int(char)]} "
 
             # Convert to regional indicator emote if letter
-            else:
+            elif char.isalpha():
                 letters += f":regional_indicator_{char}: "
 
+            # Convert to character emote
+            else:
+                letters += f"{specials[str(char)]}"
```
Comment (kdavis) [inline letters/letters.py:62]: String interpolation isn't needed here

File: letters/letters.py [hunk 2]
```diff
@@ -12,41 +15,64 @@ class Letters(commands.Cog):
...
+
         # Replace =>3 spaces with two
-        letters = re.sub(' {3,}', '  ', letters)
+        letters = re.sub(" {3,}", "  ", letters)
+
+        # Define output
+        if raw:
+            output = f"```{letters}```"
+        else:
+            output = f"{letters}"
```
Comment (kdavis) [inline letters/letters.py:71]: String interpolation isn't needed here

File: letters/letters.py [hunk 3]
```diff
@@ -12,41 +15,64 @@ class Letters(commands.Cog):
...
         input = msg.lower()
 
-        # Ensure it doesn't contain any special chars, numbers, etc.
-        regexp = re.compile(r'[^a-z0-9 ]')
+        # Check for raw flag
+        if input.startswith("-raw"):
+            raw = True
+            input = input.lstrip("-raw").lstrip()
+
+        else:
```
Comment (kdavis) [inline letters/letters.py:34]: This else could be simplified to instantiating `raw` as False above the if statement

**Review by kdavis — CHANGES_REQUESTED** [review]
Really proud of your first cog x

File: letters/letters.py [hunk 4]
```diff
@@ -3,50 +3,76 @@
...
+    elif char.isdigit():
+        return f"{nums[int(char)]} "
+
+    # Convert to regional indicator emote if letter
+    elif char.isalpha():
+        return f":regional_indicator_{char}: "
+
+    # Convert to character emote
+    else:
+        return specials[char]
```
Comment (tigattack) [inline letters/letters.py:29]: This needs to be changed back to use string interpolation with a space at the end, matching lines 21 and 25.

**Review by issy — APPROVED** [review]
With punctuation spacing now fixed I think this is merge-ready

**Review by issy — APPROVED** [review]

**Pull request merged**